src/prpy/util.py

```diff
@@ -986,6 +1184,140 @@ def GetCollisionCheckPts(robot, traj, include_start=True, start_time=0.,
+    temp_traj_cspec = traj.GetConfigurationSpecification()
+    temp_traj_cspec.AddDeltaTimeGroup()
+    temp_traj = openravepy.RaveCreateTrajectory(env, '')
+    temp_traj.Init(temp_traj_cspec)
+
+    # Set timing of first waypoint in temporary trajectory to t=0
+    waypoint = traj.GetWaypoint(0, temp_traj_cspec)
+    q0 = traj_cspec.ExtractJointValues(waypoint, robot, dof_indices)
+    delta_time = 0.0
+    temp_traj_cspec.InsertDeltaTime(waypoint, delta_time)
+    temp_traj.Insert(0, waypoint)
+
+    # Get the resolution (in radians) for each joint
+    q_resolutions = robot.GetDOFResolutions()[dof_indices]
+
+    # Iterate over each segment in the trajectory and set
+    # the timing of each waypoint in the temporary trajectory
+    # so that taking steps of t=1 will be within a required error norm.
+    for i in range(1, num_waypoints):
+        # We already have the first waypoint (q0) of this segment,
+        # so get the joint values for the second waypoint.
+        waypoint = traj.GetWaypoint(i)
+        q1 = traj_cspec.ExtractJointValues(waypoint, robot, dof_indices)
+        dq = numpy.abs(q1 - q0)
+        max_diff_float = numpy.max( numpy.abs(q1 - q0) / q_resolutions)
+
+        # Get the number of steps (as a float) required for
+        # each joint at DOF resolution
+        num_steps = dq / q_resolutions
+
+        # Calculate the norm:
+        #
+        # norm_order = 1  ==>  The L1 norm
+        # Which is like a diamond shape in configuration space
+        # and equivalent to: L1_norm=sum(num_steps)
+        #
+        # norm_order = 2  ==>  The L2 norm
+        # Which is like an ellipse in configuration space
+        # and equivalent to: L2_norm=numpy.linalg.norm(num_steps)
+        #
+        # norm_order = inf  ==>  The L_infinity norm
+        # Which is like a box shape in configuration space
+        # and equivalent to: L_inf_norm=numpy.max(num_steps)
+        norm = numpy.linalg.norm(num_steps, ord=norm_order)
+
+        # Set timing of this waypoint
+        waypoint = traj.GetWaypoint(i, temp_traj_cspec)
+        delta_time = norm
+        temp_traj_cspec.InsertDeltaTime(waypoint, delta_time)
+        temp_traj.Insert(i, waypoint)
+
+        # The last waypoint becomes the first in the next segment
+        q0 = q1
+
+    traj_duration = temp_traj.GetDuration()
+
+    # Sample the trajectory using the specified sequence
+    if sampling_order == None:
+        sampling_order = 'linear'
+    seq = None
+    if sampling_order == 'van_der_corput':
+        # An approximate Van der Corput sequence, between the
+        # start and end points
+        seq = VanDerCorputSampleGenerator(0, traj_duration, step=2)
+    elif sampling_order == 'linear':
+        # (default) Linear sequence, from start to end
+        seq = SampleTimeGenerator(0, traj_duration, step=2)
+    else:
+        error = "Unknown sampling_order '" + sampling_order + "' "
+        raise ValueError(error)
```

Comment:
I suggest that you change the `sampling_order` parameter to directly accept a generator function. The function would accept `(start, end, step)` and return a generator. It could default to  `GetLinearCollisionCheckPts` to replicate the current default behavior.